src/cat-data-service/src/cli.rs

```diff
@@ -22,7 +26,30 @@ impl Cli {
             Self::Run(settings) => {
                 logger::init(settings.log_level).unwrap();
 
-                let state = Arc::new(State::new(Some(settings.database_url)).await?);
+                let state = Arc::new(
+                    State::new_with_delay(Some(settings.database_url), settings.delay_seconds)
+                        .await?,
+                );
+
+                // Check the schema version, if connection to DB timesout, log as warning,
+                // otherwise, return Error.
+                match state.event_db.schema_version_check().await {
```

Comment:
Why you have moved this validation out from the event-db crate ?
I think this check should under the responsibility of the db itself, not the service itself ?

Reply:
@Mr-Leshiy I have reverted the responsibility to the db and created an alternative way for the service to run the schema version check.